forEach.js

```diff
@@ -11,3 +11,44 @@ function cb(item, index, arr){
 };
 
 customForEach(family, cb);
+
+// Slice
+const array = ["t", "e", "s", "r"];
+
+function slice(arr,start,end) {
+  end = end || arr.length;
+  const newArr = [];
+  for(let i = start; i < arr.length; i++){
+    newArr.push(arr[i]);
+    if (arr[i] === arr[end - 1]) return newArr ;
+  }
+};
+ 
+console.log('Slice', slice(array, 0));
+
+// IndexOf
+function indexOf(item) {
+  for (let i = 0; i <= array.length; i++) {
+    if (array[i] === item) {
+      return i;
+    } else if (array[i] === undefined) {
+      return -1;
+    }
+  }
```

Comment:
try not to check for undefined, because they might be cases when you have a valid array element with undefined value